internal/inventory/app/app.go

```diff
@@ -69,51 +67,34 @@ func NewInventoryApp() *InventoryApp {
 // Initialize will initialize the AppFunctionsSDK and Logging Client. It also reads the user's
 // configuration and sets up the API routes.
 func (app *InventoryApp) Initialize() error {
-	app.edgexSdk = &appsdk.AppFunctionsSDK{ServiceKey: serviceKey}
-	err := app.edgexSdk.Initialize()
-	app.lc = app.edgexSdk.LoggingClient // ensure logging client is assigned before returning
-	if err != nil {
-		return errors.Wrap(err, "SDK initialization failed")
+	var ok bool
+	var err error
+
+	app.service, ok = pkg.NewAppService(serviceKey)
+	if !ok {
+		return errors.New("Failed to create application service")
 	}
 
+	app.lc = app.service.LoggingClient() // ensure logging client is assigned before returning
+
 	app.lc.Info("Starting.")
 
-	sdkFlags := getSdkFlags()
+	app.confUpdateCh = make(chan interface{})
 
-	appSettings := app.edgexSdk.ApplicationSettings()
-	if appSettings == nil {
-		return errors.New("missing application settings")
-	}
-	if app.configClient, err = getConfigClient(sdkFlags); err != nil {
-		return errors.Wrap(err, "failed to create config client")
+	app.config = inventory.ServiceConfig{}
```

Comment:
I'm not really sure this is needed, unless you are wanting to explicitly clear/reset it, it should be auto initialized by Go.

Reply:
@ajcasagrande , this is require for app.service.LoadCustomConfig below.

Reply:
yes, agreed will remove